docs/source/eventproc.rst

```diff
@@ -0,0 +1,201 @@
+Event Processing
+=========================
+
+Overview
+--------
+
+The Event Processing library allows you to run arbitrary code (event handlers) at predefined events (also called hooks) within the PredictiveGrid platform.  A sample use case might be if you would like to upload new COMTRADE files using the COMTRADE ingress and have a transform or analytical process run immediately after import.
+
+Notifications
+~~~~~~~~~~~~~~~~~~~
+
+After your event handler is executed, an email is sent to the success or failure email address supplied during registration.  If successful, the email will contain any text that was returned by your Python function.  If an exception is raised, then the handler code is regarded as unsuccesful and a notification is sent to the corresponding failure email address with exception details.
+
+Dependencies
+~~~~~~~~~~~~~~~~~~~
+
+Our platform will provide a base python 3.7.3 environment with which you can add any desired libraries installable with `pip`.  When registering an event handler you can provide the content for a requirements file as if you had run `pip freeze` at the command line.  We provide some standard libraries such as:
+
+* btrdb
+* matplotlib
+* numpy
+* pandas
+* ray
+* scikit-learn
+* scipy
+* statsmodels
+* tensorflow
+* torch
+* torchvision
+
+Flags
+~~~~~~~~~~~~~~~~~~~
+
+Each event handler also includes a `flags` argument.  This allows the user to choose which event handlers should run in any given scenarion.  For instance, if submitting new COMTRADE files to the ingress, you can choose the "voltage-transform" flag to indicate that any event handlers registered with this flag will be executed.  Multiple flags can be added to your event handler, and users can choose from multiple flags to determine which handlers should run.  In the latter case, a logical OR is used such that any handler with at least one of the chosen flags will be scheduled for execution.
+
+Connections
+-----------
+
+Each of the following calls require a `Connection` object (different from the btrdb library connection object) as the first argument.  When registering a new event handler, the API key in the `Connection` object will be used as the execution time security context - meaning all code will be executed as the registering user.
```

Comment:
Just a nit -- to avoid confusing developers, is there any other name we can use? EventContext, maybe?

Reply:
Hmm.. I'm honestly not sure how I feel about this.  I felt `Connection` was a natural name but don't want to create confusion around it if you think it will do so..  

I'm going to leave in for now partly  because there is a massive replace required for this and I'd like to avoid that testing due to time.